### PR TITLE
PN-11897 - Rework legal contacts components

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Rework legal contacts components on PF and PG

## List of changes proposed in this pull request
- Create PecContactItem component to handle editing, inserting, and validating contacts
- Removed InsertLegalContact component

## How to test
On PF and PG try to insert, remove and edit a legal contact

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/components/Contacts/LegalContacts.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const title = (
+    <Grid container spacing={1} alignItems="flex-end" direction="row">
+      <Grid item xs="auto">
+        {t('legal-contacts.subtitle-2', { ns: 'recapiti' })}
+      </Grid>
+    </Grid>
+  );
+
+  const defaultAddress = legalAddresses.find((a) => a.senderId === 'default');
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/LegalContacts.tsx:25]: Call it pecAddress and add the check on channelType in the find function

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/LegalContacts.test.tsx [hunk 2]
```diff
@@ -297,32 +292,4 @@ describe('LegalContactsList Component', async () => {
       expect(result.container).not.toHaveTextContent(defaultAddress!.value);
     });
   });
-
-  it('checks invalid pec', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/LegalContacts.test.tsx:301]: add the test "renders component - no contacts", copied from InsertLegalContact.test.tsx - renders InsertLegalContact

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/PecContactItem.test.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { digitalLegalAddresses } from '../../../__mocks__/Contacts.mock';
+import { RenderResult, act, fireEvent, render, waitFor } from '../../../__test__/test-utils';
+import { DigitalContactsCodeVerificationProvider } from '../DigitalContactsCodeVerification.context';
+import PecContactItem from '../PecContactItem';
+
+describe('PecContactItem component', () => {
+  let result: RenderResult;
+
+  it('type in an invalid pec', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/PecContactItem.test.tsx:9]: add the test type in a valid pec copied from InsertLegalContact.test.tsx - checks valid pec

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/LegalContacts.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    <DigitalContactsCard
+      sectionTitle={t('legal-contacts.title', { ns: 'recapiti' })}
+      title={title}
+      subtitle={t('legal-contacts.description', { ns: 'recapiti' })}
+      avatar={<IllusEmailValidation />}
+    >
+      <PecContactItem
+        value={legalAddresses.find((a) => a.senderId === 'default')?.value ?? ''}
+        blockDelete={legalAddresses.length > 1}
+        verifyingAddress={defaultAddress ? !defaultAddress.pecValid : false}
```
[the remark this answers is not in the document]
Comment (alessandrogelmi) [reply packages/pn-personafisica-webapp/src/components/Contacts/LegalContacts.tsx:37]: decided to keep the ternary

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**